lib/astarte_rpc/config.ex

```diff
@@ -21,27 +21,76 @@ defmodule Astarte.RPC.Config do
   This module helps the access to the runtime configuration of Astarte RPC
   """
 
-  @doc """
-  Returns the amqp_connection options or an empty list if they're not set.
+  alias Astarte.RPC.Config.AMQPQueueArguments
+  use Skogsra
+
+  @envdoc "Username for accessing the AMQP broker."
+  app_env :amqp_connection_username, :astarte_rpc, :amqp_connection_username,
+    os_env: "RPC_AMQP_CONNECTION_USERNAME",
+    type: :binary,
+    default: "guest"
+
+  @envdoc "Password for accessing the AMQP broker."
+  app_env :amqp_connection_password, :astarte_rpc, :amqp_connection_password,
+    os_env: "RPC_AMQP_CONNECTION_PASSWORD",
+    type: :binary,
+    default: "guest"
+
+  @envdoc "The hostname or IP of the AMQP broker."
+  app_env :amqp_connection_host, :astarte_rpc, :amqp_connection_host,
+    os_env: "RPC_AMQP_CONNECTION_HOST",
+    type: :binary,
+    default: "localhost"
+
+  @envdoc "The Virtual Host to be used in the AMQP broker. Must be the same for all components."
+  app_env :amqp_connection_virtual_host, :astarte_rpc, :amqp_connection_virtual_host,
+    os_env: "RPC_AMQP_CONNECTION_VIRTUAL_HOST",
+    type: :binary,
+    default: "/"
+
+  @envdoc "The port of the AMQP broker to connect to."
+  app_env :amqp_connection_port, :astarte_rpc, :amqp_connection_port,
+    os_env: "RPC_AMQP_CONNECTION_PORT",
+    type: :integer,
+    default: 5672
+
+  @envdoc "The prefetch count of the AMQP connection. A prefetch count of 0 means unlimited (not recommended)."
+  app_env :amqp_prefetch_count, :astarte_rpc, :amqp_prefetch_count,
+    os_env: "RPC_AMQP_PREFETCH_COUNT",
+    type: :integer,
+    default: 300,
+    required: true
+
+  @envdoc """
+  Max length of the server AMQP queue. If 0 the queue will be unbounded, otherwise it will be limited to tha    t length and new publishes will be dropped while the queue is full. WARNING: changing this value requires manually     deleting the queue
   """
-  def amqp_options do
-    Application.get_env(:astarte_rpc, :amqp_connection, [])
-  end
+  app_env :amqp_queue_max_length, :astarte_rpc, :amqp_queue_max_length,
+    os_env: "RPC_AMQP_QUEUE_MAX_LENGTH",
+    type: :integer,
+    default: 0
+
+  @envdoc "The AMQP queue arguments."
+  app_env :amqp_queue_arguments, :astarte_rpc, :amqp_queue_arguments,
+    os_env: "RPC_AMQP_QUEUE_MAX_LENGTH",
+    type: AMQPQueueArguments,
+    default: []
```

Comment:
I don't think this needs to be a Skogsra type, this could be just a fuction like `amqp_options!` that calls `amqp_queue_max_length`